### PR TITLE
buildFHSUserEnv: add version arg

### PR DESCRIPTION
**Pull request by cdepillabout**
###### Description of changes

This PR adds an optional `version` argument to `buildFHSUserEnvChroot` and `buildFHSUserEnvBubblewrap`.  This makes it so that derivations produced by these functions can have their versions set.  This can help end-users more clearly see the version of the packages they are using.

For instance, on current `master`:

```console
$ nix-build -E 'with import ./. {}; buildFHSUserEnvChroot { name = "hello"; }'
/nix/store/1wxqalaknmjsdhd576377wxjl7rn9zkh-hello
$ nix-build -E 'with import ./. {}; buildFHSUserEnvBubblewrap { name = "hello"; }'
/nix/store/nn6gvq4gvz64nhcrs8si08cbqny1n2pq-hello
```

With this PR, using the new `version` argument:

```console
$ nix-build -E 'with import ./. {}; buildFHSUserEnvChroot { name = "hello"; version = "1.2.3"; }'
/nix/store/vzg1rdkqdxk00dx3li03f3h3l04qwaab-hello-1.2.3
$ nix-build -E 'with import ./. {}; buildFHSUserEnvBubblewrap { name = "hello"; version = "1.2.3"; }'
/nix/store/vs0z3cbwbmw11wqz6ykrah1sc2vna68g-hello-1.2.3
```

This was suggested in https://github.com/NixOS/nixpkgs/pull/218162#issuecomment-1445037235.

Derivations created from `buildFHSUserEnv` not having versions by default has confused at least two end users (myself included):

https://discourse.nixos.org/t/how-to-run-latest-anki-download/25406/5?u=cdepillabout

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bjornfor — APPROVED** [review]

**Review by Atemu — CHANGES_REQUESTED** [review]
Some minor nits, going to test it later.

File: pkgs/build-support/build-fhs-userenv-bubblewrap/default.nix [hunk 1]
```diff
@@ -192,7 +194,11 @@ let
 
   bin = writeShellScriptBin name (bwrapCmd { initArgs = ''"$@"''; });
 
-in runCommandLocal name {
+  versionStr = lib.optionalString (! isNull version) ("-" + version);
```
[the remark this answers is not in the document]
Comment (cdepillabout) [reply pkgs/build-support/build-fhs-userenv-bubblewrap/default.nix]: Good call, much nicer.

**Comment by cdepillabout** [comment]
@Atemu pinging you in case you get a chance to review this

**Comment by Atemu** [comment]
Totally forgot to merge this, should have been my first action! I'll take a look later hopefully.

**Comment by cdepillabout** [comment]
@Atemu just another ping for a review on this.

Alternatively, we could go ahead and merge it, and if you later get some time to review, we could fix anything in a follow-up PR.

**Review by Atemu — APPROVED** [review]

**Pull request merged**